### PR TITLE
Add Code First Migration and update changes

### PR DESCRIPTION
**Pull request by M-HAMMAD-HAZELSOFT**

### CODE REVIEW

File: Migrations/20230606071248_initialCreate.cs [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+                    Password = table.Column<string>(type: "nvarchar(max)", nullable: false)
+                },
+                constraints: table =>
+                {
+                    table.PrimaryKey("PK_users", x => x.Id);
+                });
+        }
+
+        /// <inheritdoc />
+        protected override void Down(MigrationBuilder migrationBuilder)
```
Comment (HassanMustafaHazel) [inline Migrations/20230606071248_initialCreate.cs:30]: check out the migration revert process again.

File: Program.cs [hunk 2]
```diff
@@ -1,7 +1,14 @@
+using Microsoft.EntityFrameworkCore;
 using webapi.Services.UserService;
+using webapi.Data;
 
 var builder = WebApplication.CreateBuilder(args);
 
+var connectionString = builder.Configuration.GetConnectionString("DefaultConnection");
+
+// Configure the DataContext with the provided connection string
+builder.Services.AddDbContext<DataContext>(opt => opt.UseSqlServer(connectionString));
```
Comment (HassanMustafaHazel) [inline Program.cs:10]: need to look into concept on dependency injection again.

File: Services/UserService/UserService.cs [hunk 3]
```diff
@@ -25,17 +28,21 @@ public UserService(IMapper mapper)
         /// <returns>A list of all users.</returns>
         public async Task<List<UsersDto>> GetAllUsers()
         {
-            return (users.Select(c => _mapper.Map<UsersDto>(c))).ToList();
+            List<Users> user = await _context.Users.ToListAsync();
+
+            return (user.Select(c => _mapper.Map<UsersDto>(c))).ToList();
```
Comment (HassanMustafaHazel) [inline Services/UserService/UserService.cs]: need to hide the password in an authentic way not to ignore the password through an auto mapper profile.

File: Services/UserService/UserService.cs [hunk 4]
```diff
@@ -69,16 +75,21 @@ public async Task<List<UsersDto>> AddUser(UsersDto newUser)
...
             if (user != null)
             {
                 // Update the user 
                 user.Name = updatedUser.Name;
                 user.Email = updatedUser.Email;
                 user.Password = updatedUser.Password;
+
+                _context.Users.Update(user);
+                await _context.SaveChangesAsync();
+
```
Comment (HassanMustafaHazel) [inline Services/UserService/UserService.cs:85]: move this mapping to controller.

**Review by HassanMustafaHazel — CHANGES_REQUESTED** [review]

File: Services/UserService/UserService.cs (hunk 4 above)
Comment (M-HAMMAD-HAZELSOFT) [reply Services/UserService/UserService.cs:85]: Done

File: Services/UserService/UserService.cs (hunk 3 above)
Comment (M-HAMMAD-HAZELSOFT) [reply Services/UserService/UserService.cs]: Done

File: Migrations/20230606071248_initialCreate.cs (hunk 1 above)
Comment (M-HAMMAD-HAZELSOFT) [reply Migrations/20230606071248_initialCreate.cs:30]: Done

File: Program.cs (hunk 2 above)
Comment (M-HAMMAD-HAZELSOFT) [reply Program.cs:10]: Done

**Review by HassanMustafaHazel — APPROVED** [review]

**Pull request merged**